arch/x86/core/x86_mmu.c

```diff
@@ -817,6 +817,27 @@ static void add_mmu_region(struct x86_page_tables *ptables,
 	}
 }
 
+
+void z_x86_add_mmu_region(uintptr_t addr, size_t size, u64_t flags,
+			  bool user_table)
+{
+	struct mmu_region rgn = {
+		.address = addr,
+		.size = size,
+		.flags = flags,
+	};
+
+	if (!user_table) {
```

Comment:
are you sure this is right?
the code in `z_x86_paging_init()` calls add_mmu_region() on z_x86_user_ptables unconditionally for all MMU_BOOT_REGIONS


```
	Z_STRUCT_SECTION_FOREACH(mmu_region, rgn) {
		add_mmu_region(&z_x86_kernel_ptables, rgn, false);			add_mmu_region(&z_x86_kernel_ptables, rgn, false);
#ifdef CONFIG_X86_KPTI	#ifdef CONFIG_X86_KPTI
		add_mmu_region(&z_x86_user_ptables, rgn, true);			add_mmu_region(&z_x86_user_ptables, rgn, true);
#endif	#endif
	}		}
```

Reply:
Fixed this one.